src/AggregateFunctions/Streaming/CountedValueMap.h

```diff
@@ -304,18 +335,22 @@ class CountedValueMap
 
     void eraseExtraElements()
     {
+        /// Only can be used in ordered map
         if (max_size <= 0)
             return;
 
-        while (m.size() > max_size)
+        if constexpr (std::is_same_v<Map, std::map<T, uint32_t, Compare>> || std::is_same_v<Map, absl::btree_map<T, uint32_t, Compare>>)
         {
-            auto last_elem = --m.end();
-            arena->free(last_elem->first);
-            m.erase(last_elem);
+            while (m.size() > max_size)
+            {
+                auto last_elem = --m.end();
+                arena->free(last_elem->first);
+                m.erase(last_elem);
+            }
         }
```

Comment:
We still need to erase hashmap elements.

Reply:
It seems that we need to implement the logic of btree map and hash map separately.